plugins/trace_api_plugin/include/eosio/trace_api/compressed_file.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include <ios>
+#include <fc/io/cfile.hpp>
+
+namespace eosio::trace_api {
+
+   class compressed_file_datastream;
+   struct compressed_file_impl;
+   /**
+    * wrapper for read-only access to a compressed file.
+    * compressed files support seeking and reading
+    *
+    * the efficiency of seeking is lower than that of an uncompressed file as each seek translates to
+    *  - 2 seeks + 1 read to load and process the seek-point-mapping
+    *  - potentially a read/decompress/discard of the data between the seek point and the requested offset
+    *
+    * More seek points can lower the average amount of data that must be read/decompressed/discarded in order
+    * to seek to any given offset.  However, each seek point has some effect on the file size as it represents a
+    * flush of the compressor which can degrade compression performance.
+    */
```

Comment:
I'd like either a reference here to the compressed file format being documented before the process method or that moved here.  Advantage of having it there is it is more likely to be maintained if it changes, but I kept thinking it would be good to have that documented somewhere and then finally found it when I got to the process method.